src/common.speech/SynthesisContext.ts

```diff
@@ -50,6 +51,9 @@ export class SynthesisContext {
                     wordBoundaryEnabled: (!!this.privSpeechSynthesizer.wordBoundary),
                 },
                 outputFormat: this.privAudioOutputFormat.requestAudioFormatString,
+            },
+            language: {
+                autoDetection: this.privSpeechSynthesizer.properties.getProperty(PropertyId.SpeechServiceConnection_AutoDetectSourceLanguages) === AutoDetectSourceLanguagesOpenRangeOptionName
```

Comment:
Please make autoDetection a boolean get method from SpeechSynthesizer to avoid SynthesisContext having knowledge of propertyIds. This line should look like:

autoDetection: this.privSpeechSynthesizer.autoDetection

(This also means SynthesisContext doesn't need to import AutoDetectSourceLanguagesOpenRangeOptionName)

Reply:
agree, will update, thx